tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala

```diff
@@ -98,7 +100,38 @@ class AppFilterImpl(
     } else {
       appNameFiltered
     }
-    appTimeFiltered.map(_.eventlog).toSeq
+    val appCriteriaFiltered = if (appArgs.filterCriteria.isSupplied && filterCriteria.nonEmpty) {
+      if (filterCriteria.endsWith("-overall")) {
+        val filteredInfo = filterCriteria.split("-")
+        val numberofEventLogs = filteredInfo(0).toInt
+        val criteria = filteredInfo(1)
+        val filtered = if (criteria.equals("oldest")) {
+          appTimeFiltered.toSeq.sortBy(_.appInfo.get.startTime).take(numberofEventLogs)
+        } else {
+          appTimeFiltered.toSeq.sortBy(_.appInfo.get.startTime).reverse.take(numberofEventLogs)
+        }
+        filtered
+      } else if (filterCriteria.endsWith("-per-app-name")) {
+        val distinctAppNameMap = apps.groupBy(_.appInfo.get.appName)
```

Comment:
so this one isn't using the other filters above? I would think the start time would at least apply, I think we could even have the app name apply as well and just of those it would take the newest.  We may want to document this as well just saying this is applied after the others

Reply:
Thanks! It was unintentional. I had missed updating this.